spec.html

```diff
@@ -6862,24 +6862,20 @@
 
     <!-- es6num="9.2.3" -->
     <emu-clause id="sec-functionallocate" aoid="FunctionAllocate">
-      <h1>FunctionAllocate (_functionPrototype_, _strict_ [,_functionKind_] )</h1>
-      <p>The abstract operation FunctionAllocate requires the two arguments _functionPrototype_ and _strict_. It also accepts one optional argument, _functionKind_. FunctionAllocate performs the following steps:</p>
+      <h1>FunctionAllocate (_functionPrototype_, _strict_, _functionKind_)</h1>
+      <p>The abstract operation FunctionAllocate requires the three arguments _functionPrototype_, _strict_ and _functionKind_. FunctionAllocate performs the following steps:</p>
       <emu-alg>
         1. Assert: Type(_functionPrototype_) is Object.
-        1. Assert: If _functionKind_ is present, its value is either `"normal"`, `"non-constructor"` or `"generator"`.
-        1. If _functionKind_ is not present, let _functionKind_ be `"normal"`.
-        1. If _functionKind_ is `"non-constructor"`, then
-          1. Let _functionKind_ be `"normal"`.
-          1. Let _needsConstruct_ be *false*.
-        1. Else let _needsConstruct_ be *true*.
+        1. Assert: _functionKind_ is either `"normal"`, `"non-constructor"` or `"generator"`.
+        1. If _functionKind_ is `"normal"`, let _needsConstruct_ be *true*.
+        1. Else, let _needsConstruct_ be *false*.
+        1. If _functionKind_ is `"non-constructor"`, let _functionKind_ be `"normal"`.
         1. Let _F_ be a newly created ECMAScript function object with the internal slots listed in <emu-xref href="#table-27"></emu-xref>. All of those internal slots are initialized to *undefined*.
         1. Set _F_'s essential internal methods to the default ordinary object definitions specified in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>.
         1. Set _F_'s [[Call]] internal method to the definition specified in <emu-xref href="#sec-ecmascript-function-objects-call-thisargument-argumentslist"></emu-xref>.
         1. If _needsConstruct_ is *true*, then
           1. Set _F_'s [[Construct]] internal method to the definition specified in <emu-xref href="#sec-ecmascript-function-objects-construct-argumentslist-newtarget"></emu-xref>.
-          1. If _functionKind_ is `"generator"`, set the [[ConstructorKind]] internal slot of _F_ to `"derived"`.
-          1. Else, set the [[ConstructorKind]] internal slot of _F_ to `"base"`.
-          1. NOTE Generator functions are tagged as `"derived"` constructors to prevent [[Construct]] from preallocating a generator instance. Generator instance objects are allocated when EvaluateBody is applied to the |GeneratorBody| of a generator function.
+          1. Set the [[ConstructorKind]] internal slot of _F_ to `"base"`.
```

Comment:
This is somewhat confusing to me since generators don't depend on this slot at all. While not necessary, would you support prepending "If _functionKind_ is not `"generator"` to this line?